### DISCUSSION TITLE
How do I contribute in UI design?

### DISCUSSION THREAD

**Discussion by YouEsBee** [ravynsoft/ravynos#179] [Q&A]
I would like to try designing some elements of the AiryxOS to make it more intuitive and user friendly. Is there any way to help?

**Comment by mszoek** [comment]
Yeah, absolutely. What did you have in mind?

If you're on the Discord we can also chat there.

**Reply by YouEsBee** [reply]
An OS with icons and interface that we are familiar with. Also, it should have its unique identity with features of both Windows and Mac, with the aesthetics of macOS. Here's a sample icon I have made for AiryxOS.
![airyx_icon](https://user-images.githubusercontent.com/98386186/151097442-103ed90d-2c38-4b01-8299-667ed1454a24.png)

**Reply by mszoek** [reply]
Cute :) Not sure about the glasses... they seem too big, maybe? Maybe just me. 
Why features of Windows - just curious?

**Reply by YouEsBee** [reply]
Features like the ability to pin apps that we have installed on the desktop and hiding all windows to expose the desktop with one click are features missing from macOS, for which I think will be useful for many. 
How about I try making the glasses smaller? I was thinking that the glasses symbolises that it is a nerd for finding files in the system :)

**Reply by vzool** [reply]
@mszoek  Aha, a nerd likes most of The Librarians with glasses. @YouEsBee What about innocent eyes in the middle of the glasses?

**Reply by YouEsBee** [reply]
yeah! I have included it in my latest 'build' :)
![airyx_nerd](https://user-images.githubusercontent.com/98386186/151524544-fb616d1b-4925-463d-b95c-3e15d6bde5ff.png)
)

**Reply by mszoek** [reply]
This is great :)

**Reply by mszoek** [reply]
Can we try one with all black face instead of going to white on the darker background? Just to compare.

**Reply by YouEsBee** [reply]
Here it is!

![airyx_nerd](https://user-images.githubusercontent.com/98386186/151639504-86d95f26-837c-4439-a369-25dac0563202.png)

**Reply by YouEsBee** [reply]
This is my airyxOS concept

![desktop preview](https://user-images.githubusercontent.com/98386186/151655661-9166398c-da13-4e93-b2f4-4b470fc69423.png)

**Reply by mszoek** [reply]
I like it!! This is very clean and modern looking. Are you willing to help us develop the UI elements, icons, etc to make this a reality?

**Reply by mszoek** [reply]
Oh also I really prefer the all-black Finder icon! Very nice.

**Reply by YouEsBee** [reply]
> I like it!! This is very clean and modern looking. Are you willing to help us develop the UI elements, icons, etc to make this a reality?

Sure!

**Reply by mszoek** [reply]
Great! Where should we start? Window decorations? Icon set?

**Reply by YouEsBee** [reply]
I have prepared a few icons so maybe we can start on icon sets first

**Reply by YouEsBee** [reply]
Hi how else can I help in the UI design?

**Reply by mszoek** [reply]
There is lots still to do :)

We need a full trash can icon as well, and the new release needs an updated Installer icon. I hacked up the Tanuki one - maybe you can make us a better one that we can use permanently. Here is the crappy one we have now: https://github.com/mszoek/airyx/blob/main/CoreServices/Installer/Icon.png. Some logo assets are here if you want them: https://github.com/mszoek/airyx/tree/main/Logo_Assets

KDE window decorations - dark and light (title bar, button images, etc). Qt/KDE and GTK themes with all the GUI widgets matching your concept art. Color palette matching the concept art (a table of hex codes and description of what each is used for)

**Reply by YappLeCunt** [reply]
looks childish IMHO

**Reply by RandomDSdevel** [reply]
@HydraJedi:  

> looks childish IMHO

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Future theming support's still being thought about.

**Reply by mszoek** [reply]
That and it's not a given we will use all these. Some I like more than others.

**Comment by YouEsBee** [comment]
![airyx_app_lib](https://user-images.githubusercontent.com/98386186/151899582-85fef6fd-4ffb-4839-9376-86f682aaf8c5.png)
![airyx_appstore](https://user-images.githubusercontent.com/98386186/151899586-bb100ce1-2e5b-4c21-85ec-c130e9c0315f.png)
![airyx_bin](https://user-images.githubusercontent.com/98386186/151899595-b2e4b035-64d8-48da-af82-341c5bad8817.png)
![airyx_browser](https://user-images.githubusercontent.com/98386186/151899596-84fe8327-1944-45f1-acf6-e87fb461917b.png)
![airyx_calendar](https://user-images.githubusercontent.com/98386186/151899598-34d10428-d4f7-499e-ba9b-bd3822274010.png)
![airyx_downloads](https://user-images.githubusercontent.com/98386186/151899599-a4e64bfc-6e90-4817-aa39-296f474b6195.png)
![airyx_files](https://user-images.githubusercontent.com/98386186/151899602-9dbdb3ff-b1b9-4e03-a218-2e69a03e2817.png)
![airyx_mail](https://user-images.githubusercontent.com/98386186/151899603-434c7534-5606-49bf-bcb8-2c620ad2806a.png)
![airyx_music](https://user-images.githubusercontent.com/98386186/151899604-cfc3f4d3-0b5d-4601-a445-7acf7768c582.png)
![airyx_nerd](https://user-images.githubusercontent.com/98386186/151899605-3af27fb1-40e0-4231-a5d6-c9fa57e53372.png)
![airyx_notes](https://user-images.githubusercontent.com/98386186/151899608-20343df8-49df-4f66-8d76-c8e8040d8548.png)
![airyx_photos](https://user-images.githubusercontent.com/98386186/151899609-ed8a6926-cc6a-4622-9859-e2d037142fd1.png)
![airyx_settings](https://user-images.githubusercontent.com/98386186/151899610-59dcadae-b615-46cf-82dc-229a6ee711c1.png)
![airyx_contacts](https://user-images.githubusercontent.com/98386186/151899967-ca555cab-3317-4328-bd3b-96ec9429ead0.png)
![airyx_photobooth](https://user-images.githubusercontent.com/98386186/151905340-c03ab1b0-c160-408e-b4e6-7b45c82022ab.png)
![airyx_terminal](https://user-images.githubusercontent.com/98386186/151905895-13f56bcf-3cf5-416e-8994-29228724df2a.png)
![airyx_txt](https://user-images.githubusercontent.com/98386186/151906895-42e69f08-eeb7-495a-887f-9ccaefefca90.png)
![airyx_player](https://user-images.githubusercontent.com/98386186/151907434-0d0abede-1d92-4c94-8108-a07631adaad1.png)
![calculator](https://user-images.githubusercontent.com/98386186/151908266-350379c7-30b6-4e03-8393-dfeb4ea3066a.png)
![voice_recorder](https://user-images.githubusercontent.com/98386186/151908691-700c80ab-8883-4e8c-8b42-2b1aeecc4250.png)

**Reply by mszoek** [reply]
Awesome. The 'app store', music and calendar may be too close to Apple's icons. Can you try something slightly different for those 3?

**Reply by YouEsBee** [reply]
sure!

**Reply by YouEsBee** [reply]
![airyx_appstore](https://user-images.githubusercontent.com/98386186/151911397-50f5c09b-eedd-4bed-862a-267003f7803d.png)
![airyx_music](https://user-images.githubusercontent.com/98386186/151969609-0f292aa8-7be7-425b-8d1a-35c13280a432.png)
![airyx_calendar](https://user-images.githubusercontent.com/98386186/151913275-28dc2705-c488-4422-a37d-16fc4412bb56.png)
Or
![airyx_calendar](https://user-images.githubusercontent.com/98386186/151973150-c6454ba7-d3a8-420b-9aae-731a293cd13d.png)

**Reply by talaviram** [reply]
Is there any reason for preferring bitmapped / rasterized over SVG / vectorized?

**Reply by mszoek** [reply]
I can work with either, but PNG has the best support.

@YouEsBee these are much better! Should be "different" enough. Can we drop the "halo" in the app store icon and try less gradient for the background? A white music note probably works better on that background. Calendar seems to have a bezel while most of the icons are flat - any reason there? Maybe we can flatten it but otherwise I love the design. Sorry to be so picky!

**Reply by YouEsBee** [reply]
These are the updated icons! 
![airyx_appstore](https://user-images.githubusercontent.com/98386186/151984913-45cf3f2e-a1b7-46ac-abad-e017c13ce4d2.png)
![airyx_music](https://user-images.githubusercontent.com/98386186/151983783-6f8d55dc-7ddc-4eba-906d-0d597a2d96ad.png)
![airyx_calendar](https://user-images.githubusercontent.com/98386186/151986961-f03993e5-b6c6-4f66-b5d4-053076d9bd91.png)

**Reply by mszoek** [reply]
Very nice, and quick work! I'm still not sure about the grid on that calendar - let me think on it.

**Reply by YouEsBee** [reply]
How about having more subtle grid lines or having just a number on the calendar page?

![airyx_calendar](https://user-images.githubusercontent.com/98386186/152071758-4ed5f65f-cb63-47e5-aa92-1e577a0ec957.png)
![airyx_calendar_number](https://user-images.githubusercontent.com/98386186/152071950-acaf7f27-2acb-4d33-ba3e-1ab244b66b2e.png)

**Reply by mszoek** [reply]
The subtle lines are better but I think what bothers me is that there are partial squares at the edges for a total of 8 squares per line. So either 6 days or 8 days? It just seems ... off. If it had 7 squares per line and the edges of those outer squares were the edges of the image, it would probably look more like a calendar to me.

But the number page looks great... how about we go with that for now?

**Reply by YouEsBee** [reply]
Alright I think its great!

**Reply by YouEsBee** [reply]
I have amended the grid icon, in times when a static icon is needed.
![airyx_calendar](https://user-images.githubusercontent.com/98386186/152073563-1c10270a-9f55-42fe-b34c-2bb33886d8a7.png)

**Comment by mszoek** [comment]
Here is a look at some of the icons at their actual size on the Dock
![image](https://user-images.githubusercontent.com/60144291/152620421-563982ae-32cd-4666-b9fc-21eeab34ad81.png)

The download folder, trash can and terminal look great.
At this size, I'm not sure I like the finder icon - something about the winking eye looks wrong. Thoughts?

**Reply by amuno34** [reply]
i would use a solid color as a background for the finder icon.

**Reply by RandomDSdevel** [reply]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;There should be a bit of a margin between icons and the edge of the Dock; that's especially noticeable on the left side with the 'Finder'/Filer icon; it shouldn't be touching the edge there.  A fix for that can come later, though.

**Reply by mszoek** [reply]
> There should be a bit of a margin between icons and the edge of the Dock; that's especially noticeable on the left side with the 'Finder'/Filer icon; it shouldn't be touching the edge there. A fix for that can come later, though.

Yep - already logged a 'todo' for that one. The section dividers got thick too - must be the new Qt library or something.

**Reply by mszoek** [reply]
Here's the project plan for Dock, BTW: https://github.com/mszoek/airyx/projects/5

**Reply by YouEsBee** [reply]
Not blinking :)
![airyx_nerd2](https://user-images.githubusercontent.com/98386186/152633609-bd21a166-71e1-48e8-8921-2d12152b76bc.png)

**Reply by YouEsBee** [reply]
![airyx_trash_full](https://user-images.githubusercontent.com/98386186/152634347-1dc37811-1715-48ce-808d-e52aefd83d7a.png)

**Reply by amuno34** [reply]
> ![airyx_trash_full](https://user-images.githubusercontent.com/98386186/152634347-1dc37811-1715-48ce-808d-e52aefd83d7a.png)
> 
> 

I would make it less defined because it will be used as a tiny icon.

**Reply by mszoek** [reply]
> Not blinking :)

It is _much_ better not blinking! Someone on Discord had a great idea too... can we try it with no glasses and using two close shades taken from the airyx.org website here? Maybe the shades of the 2 lowercase 'm's?

![image](https://user-images.githubusercontent.com/60144291/152647216-ac2319ad-f0d3-4a04-9876-788dd4d48eca.png)

**Reply by mszoek** [reply]
I think the full trash looks awesome at icon size, actually.

![image](https://user-images.githubusercontent.com/60144291/152647547-9d065693-dc40-49d1-b681-005d080fb492.png)

**Reply by YouEsBee** [reply]
With glasses, gradient A
![airyx_nerd_a1](https://user-images.githubusercontent.com/98386186/152665590-d120475e-b6e6-4869-9eba-edfaa748568c.png)
With glasses, gradient B
![airyx_nerd_a2](https://user-images.githubusercontent.com/98386186/152665591-2aa13b38-3716-4972-b9d9-8d33fc869193.png)
Without glasses, gradient A
![airyx_nerd_b1](https://user-images.githubusercontent.com/98386186/152665592-c2154d26-2574-4bd6-a6fe-de8ffc7416ec.png)
Without glasses, gradient B
![airyx_nerd_b2](https://user-images.githubusercontent.com/98386186/152665593-519c05c1-6962-4cf0-b1d0-4e23fa0d21e6.png)

**Reply by YouEsBee** [reply]
Installer
![airyx_installer_0 4](https://user-images.githubusercontent.com/98386186/152665993-6548a2eb-2ed4-44fb-bc1e-c91a011f7428.png)

**Reply by mszoek** [reply]
Hah, this is perfect!! Adorable.

**Reply by YouEsBee** [reply]
### Light Mode

**Font:**

Arial


**Dock:** 

Colour: #f4f4f4
Opacity: 55%
Distance between icons: 40px
Padding between icon and dock: 28px
Radius of dock: 50px
Diameter of circle below opened app: 15px
Width of separator: 13px


**Windows:**

Distance between ‘traffic light’ and window edge: 45px
Distance between 'traffic lights': 15px
Main window contents background colour: #f3f3f3
Description of tab bg colour: #f8f7f7
Line separating tab description and main window contents: 5px
Colour of tab outline: #dedede
Radius of tab: 50px
Outer shadow for depth of window: 21px 


**Button:**

Radius: 15px
Colour: #ffffff
Border colour: #e6e6e6


**Global Menu:**

Padding between menus: 65px
Opacity: 75%
Colour: #ffffff
Distance between icon shortcuts (e.g. wifi, bluetooth, search): 40px

**Comment by YappLeCunt** [comment]
![eeeee](https://user-images.githubusercontent.com/40585717/152674863-dfa15832-58e1-4709-a798-ad2f11c5498a.png)
how about this?

**Reply by mszoek** [reply]
This is also great! Sheesh, so much artistic talent suddenly available to this little project :) I'll try it out as well.

**Reply by mszoek** [reply]
This actually does look really nice and fits well with the rest of YouEsBee's icons. 
![image](https://user-images.githubusercontent.com/60144291/152684716-6cd68787-df9a-4939-94f8-f23978663ef9.png)

OK, let's try this set for now. Thank you both for the contributions!!

**Reply by probonopd** [reply]
IANAL but be careful not to violate trademarks

**Reply by mszoek** [reply]
Naturally. Many people have made very similar icons for Linux desktops without issue (AFAIK) but admittedly we are a bit of a special case... you think it's too similar? Apple does have the blue and graphite Finder icons registered with USPTO.

The purple gradient one is safer for sure.

**Reply by YouEsBee** [reply]
I guess it's better to not draw too much attention from Apple. We should use an icon that does not seem like a random Finder clone.

**Reply by mszoek** [reply]
Do you want to try a totally different concept for the Filer icon?

**Reply by YappLeCunt** [reply]
> I guess it's better to not draw too much attention from Apple. We should use an icon that does not seem like a random Finder clone.

+1
play it safe, you can let people change the icons later to the icons of their choice

**Reply by KevDoy** [reply]
You gotta be careful with the face on finder. Apple used to use that as the macOS logo. If it isn’t a registrered trademark, they can still establish a history of usage. 

For Trenta Icons, I did it this way.
![](https://trenta.io/img/icons-dock.png)


@YouEsBee maybe you can use these icons as a jumping off point. I plan to continue maintaining that project. Maybe Airyx can fork it?

edit: plus, they’re all SVG!
